### PR TITLE
Symmetry-transformed overlap integrals

### PR DESCRIPTION
**Pull request by thchr**
Hi @stevengj,

As I mentioned last week, I've had some code laying around for a while in my fork (and have used it a bunch) which computes the overlap integral of modes under a symmetry operation.
This allows one to easily compute which irrep a given solution transforms like by using the little group operations and character tables (e.g., via [Crystalline.jl](https://github.com/thchr/Crystalline.jl)). I figured this might be interesting to contribute back to your main repo, since it could be of general interest.

The way it works is just to do the integral in the real-space basis: ideally, it could've been done in the plane-wave basis - but it seemed simpler to do in the real-space basis since the transformation of the mode polarization is straightforward there.
It's reasonably fast as-is: I found the cost of evaluating the symmetries to be negligible compared to obtaining the band solutions themselves.

There are some caveats to consider if you think this might be worth merging:
- Currently, the functionality only works for `mpb` and not for `mpb-mpi`, `mpbi`, or `mpbi-mpi`. There are some issues in the `HAVE_MPI` and `SCALAR_COMPLEX` cases that I haven't managed to get to the bottom of.
- There's no pretty-printing of the output included here. I have some Scheme code elsewhere that achieves this (e.g., by [printing symmetry operation matrix-vector pairs as coordinate triplets](https://gist.github.com/thchr/21a359f883dbb388c74486fcf14996b8)) but I wasn't sure if it was worthwhile to include here.
- There are currently no tests; I could add some if that's desirable.

-Thomas

### CODE REVIEW

File: mpb/fields.c [hunk 1]
```diff
@@ -706,32 +706,31 @@ number get_energy_point(vector3 p)
      return f_interp_val(p, mdata, (real *) curfield, 1, 0);
 }
 
-cvector3 get_bloch_field_point(vector3 p)
+void get_bloch_field_point_(scalar_complex *field, vector3 p)
```
Comment (stevengj) [inline mpb/fields.c:709]: Maybe `scalar_complex field[3]` would be clearer here?

**Comment by stevengj** [comment]
Is there any test for this that could be included?

**Comment by thchr** [comment]
We could add a simple test that e.g. checks against regressions for a simple operation, e.g. inversion? 
I have some 2D and 3D .ctl files that do that; would that be sufficient or do we want to test more thoroughly?

File: mpb/fields.c (hunk 1 above)
Comment (thchr) [reply mpb/fields.c:709]: I wasn't aware of this syntax but that does seem better in terms of readability.
Would it make sense to make it `scalar_complex field[static 3]` to guarantee the size exaplicitly to the compiler (that's what I understood from [trying to read up on this syntax](https://stackoverflow.com/a/26410652/9911781))?

**Comment by stevengj** [comment]
Yes, that should be fine.

**Comment by thchr** [comment]
There's some tests now and I resolved the `mpb/field.c` comment as well.

File: mpb/fields.c [hunk 2]
```diff
@@ -706,7 +706,7 @@ number get_energy_point(vector3 p)
      return f_interp_val(p, mdata, (real *) curfield, 1, 0);
 }
 
-void get_bloch_field_point_(scalar_complex *field, vector3 p)
+void get_bloch_field_point_(scalar_complex field[static 3], vector3 p)
```
Comment (stevengj) [inline mpb/fields.c:709]:
```suggestion
void get_bloch_field_point_(scalar_complex field[3], vector3 p)
```

This `static` is a C99 feature, but we don't currently require C99 (which involves setting flags on some compilers, and I don't know if it's supported by all C++ compilers.

Comment (thchr) [reply mpb/fields.c]: Okay, makes sense (I had tried to look up which C versions MPB were supporting but could not immediately find out - so I just assumed C99+; though now I think I remember you telling me a similar thing before).

**Review by stevengj — APPROVED** [review]

**Comment by stevengj** [comment]
Looks good. I’m assuming `make check` passes, since GitHub is not running it automatically at the moment?

**Comment by thchr** [comment]
I checked it locally and it passed, yep.

**Pull request merged**